apiserver/facades/client/modelgeneration/modelgeneration_test.go

```diff
@@ -115,18 +116,51 @@ func (s *modelGenerationSuite) TestTrackBranchSuccess(c *gc.C) {
 	})
 }
 
-func (s *modelGenerationSuite) TestCommitGeneration(c *gc.C) {
-	defer s.setupModelGenerationAPI(c, func(ctrl *gomock.Controller, _ *mocks.MockState, mod *mocks.MockModel) {
+func (s *modelGenerationSuite) TestCommitBranchSuccess(c *gc.C) {
```

Comment:
From the API server's point of view, all the branch component application needs to be in the database tier IMO.

Reply:
Yes; it is there now.